### ISSUE TITLE
Exclude annotation is not processed when pod has multiple containers

### ISSUE THREAD

**Issue by Scukerman** [fluent/fluent-bit#2339]
## Bug Report

**Describe the bug**
`fluentbit.io/exclude` annotation doesn't work on pod if the pod has multiple containers.
`fluentbit.io/exclude_stdout-nginx` also doesn't work.

**To Reproduce**
- Enable `K8S-Logging.Exclude` in the fluentbit config.
- Annotate a pod with at least two containers with the command (`kubectl annotate po -l app=demo-server fluentbit.io/exclude='true'`, adjust it for your environment).
- Delete the pod so fluent bit could reload the configuration and apply the annotations.

**Expected behavior**
All pod logs of all containers should be excluded from processing

**Screenshots**

**Your Environment**

* Version used: 1.4.3
* Configuration: https://pastebin.com/muDaDxB4
* Environment name and version (e.g. Kubernetes? What version?): kubernetes 1.17.7
* Server type and version: bare metal
* Operating System and version: ubuntu
* Filters and plugins: kubernetes

**Comment by connorworkman** [comment]
Bug still exists as of fluent-bit v1.9.4

**Comment by Scukerman** [comment]
I wish they could fix it in 3.0.0 but I won't check it because I switched to another solution

**Comment by brownz11** [comment]
I know this is an old issue, but it's one of the first Google results.

In my case the (v2.1.0) it's not a bug with the exclude, but the buffer setting for the Kubernetes Filter was too low. So it would get a response, but discard it and have no annotations to do the filtering on. Luckily it's an easy fix...

```
    [FILTER]
        Name                kubernetes
        Match               kube.*
        Kube_URL            https://kubernetes.default.svc.cluster.local:443
        Buffer_Size         1M #Default is 32k which is was not enough for my cluster where we had sidecars
        K8S-Logging.Exclude On
```
